lib/publiccloud/instance.pm

```diff
@@ -604,6 +604,47 @@ sub network_speed_test() {
     record_info("curl $rmt_host", $self->run_ssh_command(cmd => "curl -w '$write_out' -o /dev/null -v https://$rmt_host/", proceed_on_failure => 1));
 }
 
+sub cleanup_cloudinit() {
+    my ($self) = @_;
+    $self->ssh_assert_script_run('sudo cloud-init clean --logs');
+    if (get_var('PUBLIC_CLOUD_CLOUD_INIT')) {
+        $self->ssh_assert_script_run('sudo rm /root/test_cloud-init.txt');
+        $self->ssh_assert_script_run('sudo zypper -n rm ed');
+    }
+}
+
+sub check_cloudinit() {
+    my ($self) = @_;
+
+    # cloud-init status
+    $self->ssh_script_retry(cmd => "sudo cloud-init status", timeout => 90, retry => 12, delay => 15);
+    $self->ssh_script_retry(cmd => "sudo cloud-init status --long", timeout => 90, retry => 12, delay => 15);
+
+    # cloud-id
```

Comment:
just to be on the safe side I would add here `die "Google not support cloud-init" if(is_gce());`  here

Reply:
This is always called `if (is_cloudinit_supported);` so that would be logic duplication.

Reply:
ok